graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/rulebuilder/parser/validation/action/ValidVariables.java

```diff
@@ -70,6 +78,9 @@ public ValidationResult validate(RuleBuilderStep step) {
             if (!parameterDescriptor.optional() && value == null) {
                 return new ValidationResult(true, f("Missing parameter %s", parameterName));
             }
+            if (value != null && tokenNames.contains(value.toString())) {
```

Comment:
If we are dealing with a regular value, it could take one of the token names as for strings, these will be placed in double quotes which is ok. 

The issue is problematic when a variable name uses a token name. 
Thus we should check it above in the output parameter name and below if it starts with `$`.
I just realized. Right now, the user won't trip over this in the rule builder as the variable names are always auto-generated. It was planned to let the user set custom names in the beginning and might be some time in the future, so I guess it makes sense to put the validation in already.